tests/python/test_compare.py

```diff
@@ -0,0 +1,31 @@
+import taichi as ti
+
+
+@ti.all_archs
+def test_compare():
+  a = ti.var(ti.i32)
+  ti.root.dynamic(ti.i, 256).place(a)
+  b = ti.var(ti.i32, shape=())
+  c = ti.var(ti.i32, shape=())
+  d = ti.var(ti.i32, shape=())
+
+  @ti.kernel
+  def func():
+    a[2] = 0 <= ti.append(a.parent(), [], 10) < 1
```

Comment:
I would break the test into multiple pieces.  
- `a[2] = 0 <= ti.append(a.parent(), [], 10) < 1` itself can be considered as one test (`test_no_duplicate_eval`).
- `a[3] = b < c == d` to `c == d != b < d > b >= b <= c` (`test_chain_compare`)
- It is also worth testing all 6 ops independently, so that every line of your code written is covered. (`test_compare_basics`)
- Also test the difference between, say, `<` and `<=`. (`test_compare_equality`)